pkg/operator/status.go

```diff
@@ -289,9 +291,105 @@ func (optr *Operator) syncUpgradeableStatus() error {
 		coStatus.Message = "One or more machine config pools are updating, please see `oc get mcp` for further details"
 	}
 
+	// don't overwrite status if updating or degraded
+	if !updating && !degraded {
+		if supported, v := optr.isKubeletSkewSupported(); !supported {
+			coStatus.Reason = "KubeletSkewUnsupported"
+			coStatus.Message = fmt.Sprintf("one or more nodes have an unsupported kubelet version skew. Please see `oc get nodes` for details and upgrade all nodes so that they have a kubelet Version of at least %v.", getMinimalSkewSupportNodeVersion(v))
+			mcoObjectRef := &corev1.ObjectReference{
+				Kind:      co.Kind,
+				Name:      co.Name,
+				Namespace: co.Namespace,
+				UID:       co.GetUID(),
+			}
+			optr.eventRecorder.Eventf(mcoObjectRef, corev1.EventTypeWarning, "KubeletSkewUnsupported", fmt.Sprintf("machine config pool contains nodes with an unsupported kubelet version skew with kube-apiserver version %v.", v))
+		}
+	}
+
 	return optr.updateStatus(co, coStatus)
 }
 
+// isKubeletSkewSupported checks the version skew of kube-apiserver and node kubelet version.
+// Returns false if the version skew is not supported. version skew > 2 is not supported.
+func (optr *Operator) isKubeletSkewSupported() (skewSupported bool, kubeAPIServerVersion string) {
+	skewSupported = true
+	nodes, err := optr.nodeLister.List(labels.Everything())
```

Comment:
One question here: will this pull ALL the nodes in the cluster? bc after thinking about this we only want it to pull the nodes managed by the MCO (ie nodes that are in machine config pools)?

Reply:
Great question. I think we want all the nodes in the cluster, since a version skew that goes out of bounds might leave a kubelet not able to communicate to the API server.

Reply:
The problem with this is that those nodes wouldn't be managed by the MCO and by reporting it in our status we seem to be implying that we, in fact, do.

Reply:
The more I consider this, the less it makes sense to me as it's not within the operator's mandate to consider and/or touch nodes that are not members of MCPs.

Reply:
If we don't check the non MCO nodes, not sure if other components will do the check?
What will be the result of that wrong implication? I saw the MCO manage the nodes using labels, if they use the MCO to manage the node, will they get notified of missing labels? So it will not be a big concern here.

Reply:
Back to my original question: does this check pull in non-MCO managed nodes? If it does, the concern is that the MCO only manages nodes that are members of a pool, we do not manage, upgrade, etc.. nodes that are not in pools. Verifying/managing  these non-pool nodes is very much out of scope for the operator. Reporting their status as part of our operator status creates an untrue implication that the MCO manages them/watches them generally.

Reply:
Users are generally responsible for non-MCO managed nodes in a cluster, I don't think we want that relationship to change.

Reply:
Fixed.